taskcluster/app_services_taskgraph/transforms/server_megazord.py

```diff
@@ -0,0 +1,139 @@
+
+def setup_linux_build_task(task, target, binary):
+    task['description'] = f'Build {binary} ({target})'
+    task['worker-type'] = 'b-linux'
+    task['worker'] = {
+        'max-run-time': 1800,
+        'docker-image': { 'in-tree': 'linux' },
+        'artifacts': [
+            {
+                'name': f'public/build/{binary}-{target}.zip',
+                'path': f'/builds/worker/checkouts/vcs/build/{binary}-{target}.zip',
+                'type': 'file',
+            }
+        ]
+    }
+    task['run'] = {
+        'using': 'run-commands',
+        'pre-commands': [
+            ['git', 'submodule', 'update', '--init'],
+            ['source', 'taskcluster/scripts/toolchain/setup-fetched-rust-toolchain.sh'],
+        ],
+        'commands': [
+            ['taskcluster/scripts/server-megazord-build.py', binary, target, 'build/'],
+        ],
+        'use-caches': True,
+    }
+    task['fetches'] = {
+        'toolchain': [
+            'rust',
+        ]
+    }
+
+def setup_mac_build_task(task, target, binary):
+    task['description'] = f'Build {binary} ({target})'
+    task['worker-type'] = 'b-osx'
+    task['worker'] = {
+        'max-run-time': 1800,
+        'artifacts': [
+            {
+                'name': f'public/build/{binary}-{target}.zip',
+                'path': f'checkouts/vcs/build/{binary}-{target}.zip',
+                'type': 'file',
+            }
+        ]
+    }
+    task['run'] = {
+        'using': 'run-commands',
+        'run-task-command': ["/usr/local/bin/python3", "run-task"],
+        'pre-commands': [
+            ["source", "taskcluster/scripts/setup-mac-worker.sh"],
+            ["source", "taskcluster/scripts/toolchain/setup-fetched-rust-toolchain.sh"],
+        ],
+        'commands': [
+            [ "taskcluster/scripts/server-megazord-build.py", binary, target, "build/" ]
+        ],
+    }
+    task['fetches'] = {
+        'toolchain': [
+            'rust-osx',
+        ]
+    }
+
+# Transform for the nimbus-assemble task
+#
+# This task produces a single zip file + checksum that combines the binaries from each individual
+# build task.
+assemble = TransformSequence()
+
+# server-megazord task that a server-megazord-assemble task depends on
+MegazordBuildDep = namedtuple("MegazordBuildDep", "label target")
+
+@assemble.add
+def setup_assemble_tasks(config, tasks):
+    for task in tasks:
+        # Which nimbus binary are we assembling?
+        binary = task['attributes']['megazord']
+
+        # Find server-megazord-build task dependencies for our binary.
+        build_task_deps = [
+            MegazordBuildDep(label, build_task.attributes['target'])
+            for (label, build_task) in config.kind_dependencies_tasks.items()
+            if build_task.kind == "server-megazord-build" and build_task.attributes.get('megazord') == binary
+        ]
+
+        task['dependencies'] = {
+            dep.label: dep.label
+            for dep in build_task_deps
+        }
```

Comment:
It would be possible to define these dependencies using Taskgraph's [from_deps](https://taskcluster-taskgraph.readthedocs.io/en/latest/reference/transforms/from_deps.html) transforms, but it's probably not worth changing at this point unless you were keen to keep the logic here to a minimum.

Reply:
`from_deps` could also handle the fetches section below I think:
https://taskcluster-taskgraph.readthedocs.io/en/latest/reference/transforms/from_deps.html#adding-fetches

In fact this whole file could likely be defined entirely in the task yaml, but it's a matter of preference so not something that needs fixing.